Makefile

```diff
@@ -1,27 +1,57 @@
-export GO111MODULE=on
-export TF_ACC_TERRAFORM_VERSION=1.5.7
-export SKIP_EXTERNAL_TABLE_TESTS=true
-export SKIP_SCIM_INTEGRATION_TESTS=true
-export SKIP_TABLE_DATA_RETENTION_TESTS=true
-
-BASE_BINARY_NAME=terraform-provider-snowflake
-TERRAFORM_PLUGINS_DIR=$(HOME)/.terraform.d/plugins
-TERRAFORM_PLUGIN_LOCAL_INSTALL=$(TERRAFORM_PLUGINS_DIR)/$(BASE_BINARY_NAME)
-COVERAGE_REPORT_FILE=coverage.txt
-COVERAGE_FLAGS=-coverprofile=$(COVERAGE_REPORT_FILE) -covermode=atomic
-
-help: ## display help for this makefile
-	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
-.PHONY: help
+export SKIP_EMAIL_INTEGRATION_TESTS=true
+export SKIP_EXTERNAL_TABLE_TEST=true
+export SKIP_NOTIFICATION_INTEGRATION_TESTS=true
+export SKIP_SAML_INTEGRATION_TESTS=true
+export SKIP_STREAM_TEST=true
+export BASE_BINARY_NAME=terraform-provider-snowflake
+export TERRAFORM_PLUGINS_DIR=$(HOME)/.terraform.d/plugins
+export TERRAFORM_PLUGIN_LOCAL_INSTALL=$(TERRAFORM_PLUGINS_DIR)/$(BASE_BINARY_NAME)
+
+default: help
 
 dev-setup: ## setup development dependencies
 	@which ./bin/golangci-lint || curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b ./bin v1.53.3
-	@which ./bin/reviewdog || curl -sSfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b ./bin v0.14.2
-.PHONY: dev-setup
+	cd tools && go install github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
+	cd tools && go install mvdan.cc/gofumpt
 
 dev-cleanup: ## cleanup development dependencies
 	rm -rf bin/*
-.PHONY: dev-cleanup
+
+docs: ## generate docs
+	go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs generate
+
+docs-check: docs ## check that docs have been generated
+	git diff --exit-code -- docs
+
+fmt: terraform-fmt ## Run terraform fmt and gofumpt
+	gofumpt -l -w .
+
+terraform-fmt: ## Run terraform fmt
+	terraform fmt -recursive ./examples/
+	terraform fmt -recursive ./pkg/resources/testdata/
+	terraform fmt -recursive ./pkg/datasources/testdata/
+
+help:
+	@grep -E '^[a-zA-Z0-9_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-23s\033[0m %s\n", $$1, $$2}'
+
+install: ## display help for this makefile
+	go install -v ./...
+
+lint: # See https://golangci-lint.run/
```

Comment:
can you extend the desc with something like "Run static code analysis, check formatting" and the link could be at the end in case anyone would like to read more

Reply:
done